<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro"  name="robot">
    <xacro:include filename="materials.xacro"/>
    <xacro:include filename="inertial_macros.xacro"/>
    
    <link name="base_link">
        
    </link>

    <joint name="main_stack_joint" type="fixed">
        <parent link="base_link"/>
        <child link="main_stack"/>
        <origin xyz="-0.180 0 0"/>
    </joint>

    <link name="main_stack">
        <visual>
            <origin xyz="0.125 0 0.09"/>
            <geometry>
                <box size="0.25 0.25 0.18"/>
            </geometry>
            <material name="white"/>
        </visual>
        <collision>
            <origin xyz="0.125 0 0.09"/>
            <geometry>
                <box size="0.25 0.25 0.18"/>
            </geometry>
        </collision>
        <xacro:inertial_box mass="0.5" x="0.25" y="0.25" z="0.18">
            <origin xyz="0.125 0 0.09" rpy="0 0 0"/>
        </xacro:inertial_box>
    </link>
    
    
    <!-- LEFT WHEEL LINK -->
    <joint name="left_wheel_joint" type="continuous">
        <parent link="base_link"/>
        <child link="left_wheel"/>
        <origin xyz="0 0.145 0" rpy="-${pi/2} 0 0"/>
        <axis xyz="0 0 1"/>
    </joint>

    <link name="left_wheel">
        <visual>
            <geometry>
                <cylinder radius="0.05" length="0.02"/>
            </geometry>
            <material name="blue"/>
        </visual>
        <collision>
            <geometry>
                <cylinder radius="0.05" length="0.02"/>
            </geometry>
        </collision>
        <xacro:inertial_cylinder mass="0.1" length="0.02" radius="0.05">
            <origin xyz="0 0 0" rpy="0 0 0"/>
        </xacro:inertial_cylinder>
    </link>

    <!-- RIGHT WHEEL LINK -->

    <joint name="right_wheel_joint" type="continuous">
        <parent link="base_link"/>
        <child link="right_wheel"/>
        <origin xyz="0 -0.145 0" rpy="${pi/2} 0 0"/>
        <axis xyz="0 0 1"/>
    </joint>

    <link name="right_wheel">
        <visual>
            <geometry>
                <cylinder radius="0.05" length="0.02"/>
            </geometry>
            <material name="blue"/>
        </visual>
        <collision>
            <geometry>
                <cylinder radius="0.05" length="0.02"/>
            </geometry>
        </collision>
        <xacro:inertial_cylinder mass="0.1" length="0.02" radius="0.05">
            <origin xyz="0 0 0" rpy="0 0 0"/>
        </xacro:inertial_cylinder>
    </link>


    <!-- CASTER WHEEL LINK -->
    <joint name="caster_wheel_link" type="fixed">
        <parent link="main_stack"/>
        <child link="caster_wheel"/>
        <origin xyz="0.05 0 0"/>
    </joint>

    <link name="caster_wheel">
        <visual>
            <geometry>
                <sphere radius="0.05"/>
            </geometry>
            <material name="black"/>
        </visual>
        <collision>
            <geometry>
                <sphere radius="0.05"/>
            </geometry>
        </collision>
        <xacro:inertial_sphere mass="0.1" radius="0.05">
            <origin xyz="0 0 0" rpy="0 0 0"/>
        </xacro:inertial_sphere>
    </link>
</robot>